src/basename.rs

```diff
@@ -1,26 +1,69 @@
 extern crate memchr;
-use self::memchr::memrchr;
-use path_parsing::{SEP, last_non_sep_i};
-
-pub fn basename<'a>(pth: &'a str, ext: &str) -> &'a str {
-  let name_end = (last_non_sep_i(pth) + 1) as usize;
-  // Known edge case, all '/'.
-  if !pth.is_empty() && name_end == 0 {
-    return &pth[..1];
+
+use path_parsing::{find_last_sep_pos, find_last_non_sep_pos, find_last_dot_pos};
+
+pub fn basename<'a>(path: &'a str, ext: &str) -> &'a str {
+  let bytes: &[u8] = path.as_bytes();
+  let mut left: usize = 0;
+  let mut right: usize = bytes.len();
+  if let Some(last_slash_pos) = find_last_sep_pos(bytes) {
+    if last_slash_pos == right - 1 {
+      if let Some(pos) = find_last_non_sep_pos(&bytes[..last_slash_pos]) {
+        right = pos + 1;
+      } else {
+        return "/";
```

Comment:
@danielpclark Returning the literal directly here, because the Windows version will have to be completely different, same for dirname.